### PR TITLE
fixing counting residues for 2nd molecule starting from 2

### LINKED ISSUE #369: xmltopology reader counts residues starting with 2 for the 2nd molecule
From https://groups.google.com/d/msg/votca/xzuifTrkEm4/e2STRdI7DAAJ:
```
csg_dump --top lammpsTopology.xml
...
molecule: 250 Pt beads: 11
1539 Name 2:Pt:C451 Type C Mass 12.0107 Resnr 549 Resname Pt Charge -0.0654008
1540 Name 2:Pt:C452 Type C Mass 12.0107 Resnr 549 Resname Pt Charge -0.0584797
1541 Name 2:Pt:C453 Type C Mass 12.0107 Resnr 549 Resname Pt Charge -0.0654008
1542 Name 2:Pt:H454 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0259088
1543 Name 2:Pt:H455 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0259088
1544 Name 2:Pt:H456 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
1545 Name 2:Pt:H457 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
1546 Name 2:Pt:H458 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
1547 Name 2:Pt:H459 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
1548 Name 2:Pt:H460 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
1549 Name 2:Pt:H461 Type H Mass 1.00794 Resnr 549 Resname Pt Charge 0.0229106
```
with this [lammpsTopology.xml](https://github.com/votca/csg/files/3089600/lammpsTopology.xml.txt). The name should be `1:Pt:H461`.

### PR DESCRIPTION
**Pull request by jkrajniak**
xmltopology reader counts residues starting with 2 for the 2nd molecule 

Fix #369

### CODE REVIEW

**Comment by junghans** [comment]
```
      Start  53: integration_Compare_csg_dump_output
27/87 Test  #53: integration_Compare_csg_dump_output ............***Failed    0.01 sec
Files "csg_dump.out" to "/home/votca/votca/csg/src/tools/references/methanol-water//csg_dump.out" are different.

```

I think you need to update the reference file as well.

File: src/libcsg/modules/io/xmltopologyreader.cc [hunk 1]
```diff
@@ -224,7 +224,6 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads,
     }
     resnr++;
   }
-  _mol_index++;
```
Comment (JoshuaSBrown) [inline src/libcsg/modules/io/xmltopologyreader.cc:228]: If it is meant to be the residue number shouldn't this line be changed:

    bname << _mol_index << ":" << molname << ":" << b.name;

to:

    bname << resnr << ":" << molname << ":" << b.name;

Comment (junghans) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: @JoshuaSBrown is right, it should be the string should `<local residue nr>:molname:`<atom name>. Maybe in xml reader `mol_index` and `resnr` is the same?

However in #369, `2:Pt:C451` was wrong as `C451` was part of the 1st residue in `Pt`.

File: src/tools/references/methanol-water/csg_dump.out [hunk 2]
```diff
@@ -13,6 +13,6 @@ molecule: 1 MEO beads: 1
 molecule: 2 MEO beads: 1
 1 Name 1:MEO:ME Type ME Mass 32.0424 Resnr 1 Resname MEO Charge 0
 molecule: 3 SOL beads: 1
-2 Name 2:SOL:WT Type WT Mass 18.0154 Resnr 2 Resname SOL Charge 0
+2 Name 1:SOL:WT Type WT Mass 18.0154 Resnr 2 Resname SOL Charge 0
```
Comment (junghans) [inline src/tools/references/methanol-water/csg_dump.out:16]: However, this looks correct now.

File: src/libcsg/modules/io/xmltopologyreader.cc (hunk 1 above)
Comment (jkrajniak) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: How the residues are counted for Lammps and Gromacs? I have never had to use in mapping 1:NAME, 2:NAME;

Comment (JoshuaSBrown) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: I am not sure how residues are handled by Gromacs or LAMMPS, the only place I have seen them used a lot is in pdb files. The use of residues is VOTCA is a bit misleading as internally residues are also treated as coarse grained components of a given molecule, but this is not consistent with the actual description of a residue in the literature. At times the use of a residue number is also internally treated as the same thing as a molecule number, hence the confusion. @junghans is my understanding correct?

Comment (junghans) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: We could make it consistent, if you have like. We just need to ensure multiple residues per molecule are supported.

Comment (JoshuaSBrown) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: Yes, I would like to make it consistent. We need to clarify the description of residue, coarse grained bead and molecule and stick to it. I will create an issue, but this is not a high priority.

Comment (JoshuaSBrown) [reply src/libcsg/modules/io/xmltopologyreader.cc:228]: https://github.com/votca/csg/issues/375

**Review by junghans — APPROVED** [review]

**Pull request merged**